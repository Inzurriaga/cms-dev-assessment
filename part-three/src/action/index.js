export const setListings = (listings) => ({
  type: 'SET_LISTINGS',
  listings
})