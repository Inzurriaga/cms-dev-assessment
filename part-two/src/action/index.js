export const setListings = (listings) => ({
  type: 'SET_LISTINGS',
  listings
});

export const setEvents = (events) => ({
  type: 'SET_EVENTS',
  events
});

export const setOffers = (offers) => ({
  type: 'SET_OFFERS',
  offers
});

export const filterCategory = (category) => ({
  type: 'FILTER_CATEGORY',
  category
});