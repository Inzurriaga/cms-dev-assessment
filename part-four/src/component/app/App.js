import React, { Component } from 'react';
import Container from '../container/Container';
import { connect } from 'react-redux';
import { setListings } from '../../action';

export class App extends Component {

  async componentDidMount() {
    await this.listings();
  }

  listings = async () => {
    try {
      const response = await fetch('https://sv-reqres.now.sh/api/listings?per_page=15');
      const listings = await response.json();
      let listingsData = listings.data;
      this.setPages(listingsData);
    } catch (error) {
      console.log(error);
    }
  }

  setPages = (listings) => {
    const pages = [];
    const cardsPerPage = this.randomizePageLayout();
    const randomlist = this.randomizeList(listings);
    const addNumberToCards = randomlist.map((card, index) => {
      return Object.assign({}, card, { number: index + 1});
    });
    const addGridLayout = this.assignGridLayout(addNumberToCards, cardsPerPage);
    cardsPerPage.forEach(amount => {
      const cards = addGridLayout.splice(0, amount.amount);
      pages.push(cards);
    });
    this.props.setListings(pages);
  }

  randomizeList = (listings) => {
    for (let i = listings.length - 1; i > 0; i--) {
      const r = Math.floor(Math.random() * (i + 1));
      [listings[i], listings[r]] = [listings[r], listings[i]]
    };
    return listings;
  }

  randomizePageLayout = () => {
    const cardsPerPage = [ {amount: 5, gridlayout: 0}, {amount: 5, gridlayout: 1}, {amount: 3, gridlayout: 2}, {amount: 2, gridlayout: 3}];
    for (let i = cardsPerPage.length - 1; i > 0; i--) {
      const r = Math.floor(Math.random() * (i + 1));
      [cardsPerPage[i], cardsPerPage[r]] = [cardsPerPage[r], cardsPerPage[i]]
    };
    return cardsPerPage;
  }

  assignGridLayout  = (lists, layouts) => {
    let indexPlacement = 0;
    return layouts.reduce((acc, layout) => {
      for(var i = 0; i < layout.amount; i++ ) {
        acc.push(Object.assign( {}, lists[indexPlacement], layout))
        indexPlacement++
      }
      return acc;
    }, []);
  }

  render() {
    return (
      <div className='App'>
        <Container />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setListings: listings => dispatch(setListings(listings))
});


export default connect(null, mapDispatchToProps)(App);
