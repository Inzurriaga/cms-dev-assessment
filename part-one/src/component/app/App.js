import React, { Component } from 'react';
import Container from '../container/Container';
import { connect } from 'react-redux';
import { setListings } from '../../action';

export class App extends Component {

  async componentDidMount() {
    await this.listings();
  }

  listings = async () => {
    try {
      const response = await fetch('https://sv-reqres.now.sh/api/listings');
      const listings = await response.json();
      this.props.setListings(listings);
    } catch (error) {
      console.log(error);
    }
  }

  render() {
    return (
      <div className='App'>
        <Container />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setListings: listings => dispatch(setListings(listings))
});

export default connect(null, mapDispatchToProps)(App);

