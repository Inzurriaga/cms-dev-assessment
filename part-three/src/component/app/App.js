import React, { Component } from 'react';
import Container from '../container/Container';
import { connect } from 'react-redux';
import { setListings } from '../../action';

export class App extends Component {

  async componentDidMount() {
    await this.listings();
  }

  listings = async () => {
    try {
      const response = await fetch('https://sv-reqres.now.sh/api/listings?per_page=15');
      const listings = await response.json();
      this.setPages(listings.data);
    } catch (error) {
      console.log(error);
    }
  }

  setPages = (listings) => {
    const pages = [];
    const cardsPerPage = [ 5, 5, 3, 2 ];
    const addNumberToCards = listings.map((card, index) => {
      return Object.assign({}, card, { number: index + 1})
    });
    cardsPerPage.forEach(amount => {
      const cards = addNumberToCards.splice(0, amount)
      pages.push(cards)
    });
    this.props.setListings(pages);
  }

  render() {
    return (
      <div className='App'>
        <Container />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setListings: listings => dispatch(setListings(listings))
});


export default connect(null, mapDispatchToProps)(App);
