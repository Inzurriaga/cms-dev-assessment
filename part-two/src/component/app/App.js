import React, { Component } from 'react';
import Container from '../container/Container';
import Header from '../header/Header';
import { connect } from 'react-redux';
import { setListings, setEvents, setOffers } from '../../action';


export class App extends Component {

  async componentDidMount() {
    await this.listings();
    await this.events();
    await this.offers();
  }

  listings = async () => {
    try {
      const response = await fetch('https://sv-reqres.now.sh/api/listings?per_page=6');
      const listings = await response.json();
      this.props.setListings(listings);
    } catch (error) {
      console.log(error)
    }
  }

  events = async () => {
    try {
      const response = await fetch('https://sv-reqres.now.sh/api/events?per_page=6');
      const events = await response.json();
      this.props.setEvents(events);
    } catch (error) {
      console.log(error)
    }
  }

  offers = async () => {
    try {
      const response = await fetch('https://sv-reqres.now.sh/api/offers?per_page=6');
      const offers = await response.json();
      this.props.setOffers(offers);
    } catch (error) {
      console.log(error)
    }
  }

  render() {
    return (
      <div className='App'>
        <Header />
        <Container />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setListings: listings => dispatch(setListings(listings)),
  setEvents: events => dispatch(setEvents(events)),
  setOffers: offers => dispatch(setOffers(offers))
});

export default connect(null, mapDispatchToProps)(App);

