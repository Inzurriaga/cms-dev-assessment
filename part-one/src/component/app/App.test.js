import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps }from './App';
import { setListings } from '../../action';

const mockListings = {}

describe('App', () => {

  describe('App', () => {

    let wrapper;
    beforeEach(() => {
      wrapper = shallow(
        <App />
      );
    })

    it('should have a snap shot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
    
    it('should call dispach for setListings', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setListings(mockListings);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setListings(mockListings);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});

