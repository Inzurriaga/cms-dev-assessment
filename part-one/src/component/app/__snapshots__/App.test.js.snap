// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should have a snap shot 1`] = `ShallowWrapper {}`;
